File: leetcode3/남효정/3940. Limit Occurrences in Sorted Array.py
class Solution:
    def limitOccurrences(self, nums: list[int], k: int) -> list[int]:
        num_type = sorted(list(set(nums)))
        ans = []
        for num in num_type:
            num_cnt = nums.count(num)
            if num_cnt > k:
                ans += ([num] * k)
            else:
                ans += ([num] * num_cnt)
        return ans
        

File: leetcode3/황은지/3940. Limit Occurrences in Sorted Array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var limitOccurrences = function (nums, k) {
  const count = Array(101).fill(0);
  const result = [];

  const putNumber = function (num, repeat) {
    for (let i = 0; i < repeat; i++) {
      result.push(num);
    }
  };

  for (let i = 0; i < nums.length; i++) {
    const num = nums[i];
    count[num]++;
    if (i === nums.length - 1 || num !== nums[i + 1]) {
      if (count[num] <= k) putNumber(num, count[num]);
      else putNumber(num, k);
    }
  }

  return result;
};
